.circleci/config.yml

```diff
@@ -20,7 +20,135 @@ parameters:
     type: string
     default: 18.16-browsers
 
+executors:
+  integration-tests:
+    docker:
+      - image: cimg/node:<< pipeline.parameters.node-version >>
+      - image: cimg/redis:7.0
+        command: redis-server --port 6380
+    resource_class: medium+
+    working_directory: ~/app
+
 jobs:
+  build:
+    executor:
+      name: hmpps/node
+      tag: << pipeline.parameters.node-version >>
+    steps:
+      - checkout
+      - run:
+          name: Update npm
+          command: 'sudo npm install -g npm@latest'
+      - restore_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+      - run:
+          name: Install Dependencies
+          command: npm ci --no-audit
+      - save_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+          paths:
+            - node_modules
+            - ~/.cache
+      - run:
+          command: |
+            npm run build
+      - run: # Run linter after build because the integration test code depend on compiled typescript...
+          name: Linter check
+          command: npm run lint
+      - run:
+          name: Link shell scripts
+          command: node_modules/.bin/shellcheck ./script/**/*[^utils] ./script/utils/**
+      - persist_to_workspace:
+          root: .
+          paths:
+            - node_modules
+            - build
+            - dist
+            - .cache/Cypress
+
+  unit_test:
+    parallelism: 4
+    executor:
+      name: hmpps/node
+      tag: << pipeline.parameters.node-version >>
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/app
+      - restore_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+      - run:
+          name: unit tests
+          command: |
+            TESTS=$(circleci tests glob "server/**/*.test.ts" | circleci tests split --split-by=timings)
+            npm run test:ci $TESTS
+      - run:
+          name: collect coverage data
+          command: |
+            mv ./coverage/coverage-final.json ./coverage/coverage_${CIRCLE_NODE_INDEX}.json
+      - store_test_results:
+          path: test_results
+      - store_artifacts:
+          path: test_results/unit-test-reports.html
+      - persist_to_workspace:
+          root: .
+          paths:
+            - coverage
+
+  coverage:
+    executor:
+      name: hmpps/node
+      tag: << pipeline.parameters.node-version >>
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/app
+      - run:
+          name: Check Coverage
+          command: |
+            npx nyc merge ./coverage/ ./coverage/.nyc_output
```

Comment:
I'm a bit unclear on what coverage checking jest is doing internally and what we're using istanbul for

Reply:
This merges all the coverage reports from the unit test run into one file called `.nyc_output` (which is what `nyc` looks for)

Reply:
thanks